frontend/app/mocks/node.ts

```diff
@@ -72,6 +72,51 @@ function getPreferredLanguageEntity(id: string | readonly string[]) {
   return parsedPreferredLanguage;
 }
 
+/**
+ * Retrieves list of letter entities based on the provided user ID.
+ *
+ * @param userId - The user ID to look up in the database.
+ * @returns The letter entity if found, otherwise throws a 404 error.
+ */
+function getLetterEntities(userId: string | readonly string[]) {
+  const parsedUserId = z.string().uuid().safeParse(userId);
+  const letterEntity = !parsedUserId.success
+    ? undefined
+    : db.letter.findMany({
+        where: {
+          userId: {
+            equals: parsedUserId.data,
+          },
+        },
+      });
+  if (!letterEntity) {
+    throw new HttpResponse('Letter Not found', { status: 404, headers: { 'Content-Type': 'text/plain' } });
+  }
+
+  return letterEntity;
+}
```

Comment:
Does that include if its empty? Or will we show the page here regardless?

Reply:
Fixed: 47fffce2dc8f67ae5494d7842d37f6acb35a77cc

Reply:
I hate nitpicking this since it's only a mock and we have no idea how the actual letter API will work.. but this should probably be a `400` instead of a `404`.

Reply:
Good point. Fixed: c20e06602530ac9a3fb8cf2b080a3b3fc36d0451